src/Build/BuildCheck/Infrastructure/BuildCheckManagerProvider.cs

```diff
@@ -392,6 +403,16 @@ public void ProcessEnvironmentVariableReadEventArgs(ICheckContext checkContext,
             }
         }
 
+        public void ProcessProjectImportedEventArgs(ICheckContext checkContext, ProjectImportedEventArgs projectImportedEventArgs)
```

Comment:
This event has BuildEventContext with EvalId - that you can use to attribute the imported file to the particular evaluation.

Reply:
done